extension/entitystore/factory.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+)
+
+var (
+	TypeStr, _  = component.NewType("entitystore")
+	entityStore *EntityStore
+)
+
+func GetEntityStore() *EntityStore {
+	return entityStore
+}
```

Comment:
Not a huge fan of this singleton case where it relies on the extension being created beforehand. Normally, extensions are supposed to be accessed via the `component.Host` through the `GetExtensions` function.

https://github.com/open-telemetry/opentelemetry-collector/blob/fee2daa249acdedddb6a4002875ef5f0c40fde12/config/confighttp/confighttp.go#L194

Reply:
Forgot about the logs agent part of this. Is there a potential for a race condition if the logs agent starts before the OTEL collector part of the agent?

Reply:
If LogAgent starts before OTEL, that should be fine. There is nil check in place when the returned variable is used, and a small delay in retrieving the entity information from `entitystore` is acceptable.

Reply:
Non-blocking comment. Not an issue.